Support for collectd signed and encrypted packets

marcv81 opened the pull request:
###### Required for all non-trivial PRs
- [X] Rebased/mergable
- [X] Tests pass
- [x] CHANGELOG.md updated
- [X] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
###### Required only if applicable
- [X] Provide example syntax

The collectd network plugin has 3 types of packets: plain, signed, and encrypted. Without this PR, InfluxDB only accepts plain packets. This PR adds support for signed and encrypted packets. This allows to securely interface InfulxDB and collectd over untrusted networks.

The detailed feature description is available on the collectd wiki: https://collectd.org/wiki/index.php/Networking_introduction

jwilder commented on services/collectd/service.go:
```diff
@@ -324,41 +347,38 @@ func (s *Service) writePoints() {
-	}
+// Unmarshal translates a ValueList into InfluxDB data points.
+func (s *Service) UnmarshalValueList(vl *api.ValueList) []models.Point {
+	timestamp := vl.Time.UTC() // TODO: precision error?
```
Is this losing support for high resolution time format?

marcv81 replied:
HR time is now implemented directly in the collectd/go-collectd library.
https://github.com/collectd/go-collectd/blob/master/network/parse.go#L168
https://github.com/collectd/go-collectd/blob/master/cdtime/cdtime.go#L26

The comment is bogus and relates to something else I fixed before submitting the original PR.

marcv81 replied:
HR time is now implemented directly in the collectd/go-collectd library.
https://github.com/collectd/go-collectd/blob/master/network/parse.go#L168
https://github.com/collectd/go-collectd/blob/master/cdtime/cdtime.go#L26

The comment is bogus and relates to something else I fixed before submitting the original PR.

marcv81 commented on services/collectd/config.go:
```diff
@@ -97,6 +107,12 @@ func (c *Config) WithDefaults() *Config {
 	if d.TypesDB == "" {
 		d.TypesDB = DefaultTypesDB
 	}
+	if d.SecurityLevel == "" {
```
I added a Validate() function. I tested that configs with an invalid security level prevent influxd from starting. But I believe the test to assign a default value remains valid, so no change on this particular line.

marcv81 commented:
@jwilder, thanks for taking the time to have a look at this PR. I believe I addressed all your comments. Please let me know if there is anything else.

mventurini-gws commented:
@jwilder (or others): this PR has been dormant for a while, except for a few rebases. I believe I implemented the requested changes so far. Could you please let me know if anything else is required?

jwilder commented on CHANGELOG.md:
```diff
@@ -22,6 +22,7 @@ The query language has been extended with a few new features:
 * Support `ON` for `SHOW` commands - [PR7295](https://github.com/influxdata/influxdb/pull/7295)
 * Support regex on fields keys in select clause - [PR7442](https://github.com/influxdata/influxdb/pull/7442)
 
+- [#7066](https://github.com/influxdata/influxdb/issues/7066): Add support for secure transmission via collectd.
```
Can you move this to `v1.2.0` section?

jwilder commented:
@marcv81 Just needs a rebase and move the changelog item and should be good to merge now. Thanks!

marcv81 commented:
@jwilder thanks for the feedback. I made a recent rebase and fixed the changelog. I will try to keep an eye on rebases more frequently!

jwilder commented on CHANGELOG.md:
```diff
@@ -4,6 +4,7 @@
 
 ### Features
 
+- [#7066](https://github.com/influxdata/influxdb/issues/7066): Add support for secure transmission via collectd.
```
@majst01 Don't forget to give yourself credit for this!

majst01 replied:
No prob, but dont now why i'm in the loop here ? :+1:

etano commented:
Anyway to have a release with this included in the near term? I would love this feature, but don't really want have to fork and build from source.

jwilder commented:
@etano You could use one of the current nightly builds.

etano commented:
@jwilder Perhaps I'm being ignorant, but I'm having trouble actually finding the packages for the nightly build. At least the files listed in Packages don't seem to exist:
https://repos.influxdata.com/ubuntu/dists/xenial/nightly/binary-amd64/Packages

Looking here, the most recent version looks like August:
https://repos.influxdata.com/ubuntu/pool/nightly/i/influxdb/

jwilder commented:
@rossmcdonald Are nightlies available in our package repos?

rossmcdonald commented:
@jwilder Not currently. We are reworking the repository to add them back, though, and I've removed the current references to avoid confusion. For now they need to be installed manually:

https://dl.influxdata.com/influxdb/nightlies/influxdb_nightly_amd64.deb
https://dl.influxdata.com/influxdb/nightlies/influxdb-nightly.x86_64.rpm
https://dl.influxdata.com/influxdb/nightlies/influxdb-nightly_linux_amd64.tar.gz

etano commented:
@rossmcdonald Thanks!

marcv81 commented:
Regarding the original question ("Any way to have a release with this included in the near term?"), do we have a plan with future release dates?

jwilder commented:
@marcv81 @etano it'll be in the 1.2 release and the current nightly builds.

marcv81 commented:
Sorry if the answer is trivial, but have we set a date for 1.2?
